library/core/src/main/java/com/google/android/exoplayer2/metadata/vorbis/VorbisCommentDecoder.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer2.metadata.vorbis;
+
+import androidx.annotation.Nullable;
+import com.google.android.exoplayer2.metadata.Metadata;
+import java.util.ArrayList;
+
+/** Decodes vorbis comments */
+public class VorbisCommentDecoder {
```

Comment:
nit: Please make classes final where possible, as it helps reduce our API surface. Ditto elsewhere.

Also, all other `XXXDecoder` classes in the metadata package implement `MetadataDecoder`, so this is slightly confusing. Given the data is already half decoded when it comes out of the JNI layer, it might be simpler just to merge this code into `FlacDecoderJni` or `FlacExtractor`.

Reply:
I will remove this class. I will also see how I can still run the tests without this.

Reply:
Would making `decodeVorbisComments` a `public static` method in the `StreamMetadata` (previously `StreamInfo`) be acceptable?

Reply:
You could pass the `ArrayList<String>` to the `StreamMetadata` constructor, have the constructor do the decoding, and have `StreamMetadata` hold the decoded `Metadata` directory rather than the `ArrayList<String>`.

 do the decoding as part of the `StreamMetadata` constructor, and then have `StreamMetadata` hold the decoded `Metadata` directly rather than the `ArrayList<String>`.

The test would then be building a `StreamMetadata` and checking that the metadata field after construction is as expected.